lib/elixir/lib/map_set.ex

```diff
@@ -358,6 +358,106 @@ defmodule MapSet do
   defp order_by_size(map1, map2) when map_size(map1) > map_size(map2), do: {map2, map1}
   defp order_by_size(map1, map2), do: {map1, map2}
 
+  @doc """
+  Maps the function `fun` over all elements in the set.
+
+  It returns a set where each element is the result of invoking `fun`
+  on each corresponding element in the set.
+
+  ## Examples
+
+      iex> MapSet.map(MapSet.new(1..3), fn x -> x * 2 end)
+      #MapSet<[2, 4, 6]>
+
+      iex> MapSet.map(MapSet.new(~w(a b c)a), &to_string/1)
+      #MapSet<["a", "b", "c"]>
+
+  """
+  @doc since: "1.14.0"
+  @spec map(t, (any -> any)) :: t
```

Comment:
Perhaps the typespecs could be made more specific?
```suggestion
  @spec map(t(a), (a -> b)) :: t(b) when a: value, b: value
```

Reply:
How about `@spec map(t(value), (value -> value)) :: t(value)` to be more consistent with other areas of the code?

Reply:
```suggestion
  @spec map(t(a), (a -> b)) :: t(b) when a: value, b: value
```